pkg/bearertoken/grpc.go

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package bearertoken
+
+import (
+	"context"
+	"fmt"
+
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/metadata"
+)
+
+const Key = "bearer.token"
+
+type tokenatedServerStream struct {
+	grpc.ServerStream
+	context context.Context
+}
+
+func (tss *tokenatedServerStream) Context() context.Context {
+	return tss.context
+}
+
+// getValidBearerToken attempts to retrieve the bearer token from the context.
+// It does not return an error if the token is missing.
+func getValidBearerToken(ctx context.Context, bearerHeader string) (string, error) {
```

Comment:
this function is used inconsistently and causes redundant checks. I am not sure you need it, I think you can limit to `GetBearerToken(ctx)` and `BearerTokenFromGRPCMetadata(ctx)`, and use them in whichever order is more appropriate for each interpceptor